pkg/http/api.go

```diff
@@ -233,6 +233,41 @@ func SetSecrets(host string, verifyTLS bool, apiKey string, project string, conf
 	return computed, Error{}
 }
 
+// GetSecretNames for specified project and config
+func GetSecretNames(host string, verifyTLS bool, apiKey string, project string, config string, includeDynamicSecrets bool) ([]string, Error) {
+	var params []queryParam
+	params = append(params, queryParam{Key: "project", Value: project})
+	params = append(params, queryParam{Key: "config", Value: config})
+	params = append(params, queryParam{Key: "include_dynamic_secrets", Value: strconv.FormatBool(includeDynamicSecrets)})
+
+	statusCode, _, response, err := GetRequest(host, verifyTLS, apiKeyHeader(apiKey), "/v3/configs/config/secrets/names", params)
+	if err != nil {
+		return nil, Error{Err: err, Message: "Unable to fetch secret names", Code: statusCode}
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal(response, &result)
+	if err != nil {
+		return nil, Error{Err: err, Message: "Unable to parse API response", Code: statusCode}
+	}
+
+	namesInterface, ok := result["names"].([]interface{})
```

Comment:
optional: Rather than fighting with `[]interface{}`, you might find it a little easier to marshal directly into the object shape that you're looking for:

```diff
diff --git a/pkg/http/api.go b/pkg/http/api.go
index 4b7d645..f879205 100644
--- a/pkg/http/api.go
+++ b/pkg/http/api.go
@@ -240,27 +240,15 @@ func GetSecretNames(host string, verifyTLS bool, apiKey string, project string,
 		return nil, Error{Err: err, Message: "Unable to fetch secret names", Code: statusCode}
 	}
 
-	var result map[string]interface{}
+	var result struct {
+		Names []string `json:"names"`
+	}
 	err = json.Unmarshal(response, &result)
 	if err != nil {
 		return nil, Error{Err: err, Message: "Unable to parse API response", Code: statusCode}
 	}
 
-	namesInterface, ok := result["names"].([]interface{})
-	if !ok {
-		return nil, Error{Err: fmt.Errorf("Unexpected type parsing secret names, expected []interface{}, got %T", result["names"]), Message: "Unable to parse API response", Code: statusCode}
-	}
-
-	var names []string
-	for _, value := range namesInterface {
-		v, ok := value.(string)
-		if !ok {
-			return nil, Error{Err: fmt.Errorf("Unexpected type parsing secret name, expected string, got %T", value), Message: "Unable to parse API response", Code: statusCode}
-		}
-		names = append(names, v)
-	}
-
-	return names, Error{}
+	return result.Names, Error{}
 }
 
 // UploadSecrets for specified project and config

```

Reply:
Much nicer! I went a bit boilerplate heavy, it seems.